vivisect/analysis/amd64/golang.py

```diff
@@ -0,0 +1,331 @@
+        if op is None:
+            return None, None
+
+    # The key opcode is "lea rax,[rip + immediate]", which points to
+    # the GO function runtime_mainPC (aka runtime_main).
+    ptr_va, runtime_va = parse_lea_raxriprel(vw, op, filename, get_content=True)
+    return ptr_va, runtime_va
+
+
+# Opcode sequence, where element [-6] is the important one.
+_GOLANG_AMD64_INSTRS = ['cld', 'call', 'mov', 'mov', 'mov', 'mov', 'call',
+                        'call', 'call', 'lea', 'push', 'push', 'call',
+                        'pop', 'pop']
+
+def find_golang_bblock_1(vw, basic_blocks, filename):
+    '''
+    Find the basic block of interest and return the opcode where
+    the special sequence of opcodes begins.  Return None if not found.
+
+    For this case, Vivisect interprets the entry as one large function
+    (IDA Pro may break it into 2 or 3 functions linked by jmp instructions).
+    The basic block of interest is in this function.
+    '''
+    the_bblock = None
+    for bblock in basic_blocks:
+        next_va = bblock[0]
+        instrs = []
+        while next_va < bblock[0] + bblock[1]:
+            try:
+                op = vw.parseOpcode(next_va)
+            except envi.InvalidInstruction:
+                op = None
+            if op is None:
+                return None
+            if (len(instrs) > 0) or (op.mnem == _GOLANG_AMD64_INSTRS[0]):
+                # Record instructions from the first opcode match.
+                instrs.append(op)
+            next_va += op.size
+        if len(instrs) >= len(_GOLANG_AMD64_INSTRS):
+            the_bblock = bblock
+            break
+    if the_bblock is None:
+        return None
+
+    for k in range(len(_GOLANG_AMD64_INSTRS)):
+        if instrs[k].mnem != _GOLANG_AMD64_INSTRS[k]:
+            return None
+    return instrs[len(_GOLANG_AMD64_INSTRS) - 6]
+
+
+def find_golang_bblock_2(vw, filename):
+    '''
+    Find the basic block of interest and return the address where
+    the special sequence of opcodes begins.  Return None if not found.
+
+    Some GO executables use an indirect jmp in the entry point,
+    and special logic is needed to locate the BB of interest.
+    Example:  1897d2de0837090ec350004ef2f9fa87.
+    '''
+    # Start from the entry point (we already know there is only one).
+    ep_va = vw.getEntryPoints()[0]
+
+    # Expect a function with one BB, ending with an indirect jump
+    # to the address loaded by "lea rax,[rip + immediate]".
+    basic_blocks = vw.getFunctionBlocks(ep_va)
+    if len(basic_blocks) != 1:
+        return None
+    instrs = golang_collect_opcodes(vw, basic_blocks[0])
+    if (len(instrs) < 2) or \
+       (instrs[-1].mnem != 'jmp') or \
+       (instrs[-2].mnem != 'lea'):
+        return None
+    op = instrs[-2]
+    ptr_va, _ = parse_lea_raxriprel(vw, op, filename)
+    if not ptr_va:
+        return None
+
+    # Analyze the function at the pointer if necessary.
+    if not vw.isFunction(ptr_va):
```

Comment:
So you don't need to have this check, since there's an identical check in `vw.makeFunction`